src/Illuminate/View/Compilers/ComponentTagCompiler.php

```diff
@@ -382,7 +382,9 @@ protected function guessAnonymousComponentUsingNamespaces(Factory $viewFactory,
                     return $view;
                 }
 
-                if ($viewFactory->exists($view = $this->guessViewName($componentName, $directory).'.'.Str::afterLast($componentName, '.'))) {
+                $lastViewPart = Str::afterLast(Str::afterLast($componentName, '.'), ':');
+
+                if ($viewFactory->exists($view = $this->guessViewName($componentName, $directory).'.'.$lastViewPart)) {
```

Comment:
can you keep this inline? I know it's lengthy but doesn't seem worth it to have the temporary variable.